src/body/body.rs

```diff
@@ -460,19 +469,28 @@ impl From<Cow<'static, str>> for Body {
 impl Sender {
     /// Check to see if this `Sender` can send more data.
     pub fn poll_ready(&mut self, cx: &mut task::Context<'_>) -> Poll<crate::Result<()>> {
-        match self.abort_tx.poll_canceled(cx) {
```

Comment:
Seems this was important for detecting when the receiver is dropped while the connection is waiting on the socket.